src/components/Misc/PostButtonModal.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { Dispatch, ReactElement, SetStateAction } from "react";
+import { Dialog, DialogTitle, DialogContent, Button } from "@mui/material/";
+
+const styles = {
+  dialogPosition: {
+    position: "absolute",
+    bottom: "500px",
+  },
+};
+
+type ModalProps = {
+  children: ReactElement<any, any>;
```

Comment:
Can we use a different type here? I think `ReactNode` or `PropsWithChildren` looks better. 

Also nitpick but I think `ModalProps` is a but vague, since we could have multiple different modals being used. The convention I use is the component name with "Props" at the end (eg. `PostButtonModalProps`), but we can also come up with our own

Reply:
`PropsWithChildren` would fit better than `ReactNode` or `ReactElement`, but I decided to go with `JSX.Element` instead because we already know what `children` is. If the `children` prop was unknown or dependent on other components, I think the other two would be better fitting but in this case, we can be more specific since `ComposePost` is the only child prop we are expecting in a `PostButtonModal` component

Reply:
As discussed offline, I don't really feel to strongly about which we use as long as we can avoid the use of the `any` type. We can stick with JSX.Element for now until we find a better reason to change it!